components/feature/downloads/src/main/java/mozilla/components/feature/downloads/DownloadsFeature.kt

```diff
@@ -210,6 +270,47 @@ class DownloadsFeature(
         block(Pair(state, download))
     }
 
+    /**
+     * Find all the app that can perform a download, including this app.
+     */
+    @VisibleForTesting
+    internal fun getDownloaderApps(context: Context, download: DownloadState): List<DownloaderApp> {
+        val downloadIntent = download.toIntent()
+
+        // Search for all activities that can render HTML.
+        val browsersIntent = Intent.parseUri("http://www.example.com/index.html", Intent.URI_INTENT_SCHEME)
+        val browsers = context.packageManager.queryIntentActivities(browsersIntent, 0)
+                .orEmpty()
+                .associateBy { it.activityInfo.name }
+
+        val thisApp = browsers.values
+                .firstOrNull { it.activityInfo.packageName == context.packageName }
+                ?.toDownloaderApp(context, download)
+
+        // Return only the apps that can perform a download plus this app.
+        return context.packageManager.queryIntentActivities(downloadIntent, 0)
```

Comment:
Here and above you pass in `0` as flag. In `Browsers` we seem to set the flag.. do we need that?
https://github.com/mozilla-mobile/android-components/blob/master/components/support/utils/src/main/java/mozilla/components/support/utils/Browsers.kt#L211-L215

Reply:
Fixed